apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -460,6 +460,10 @@ type Storage struct {
 	// Whether this disk represents rotational storage
 	Rotational bool `json:"rotational,omitempty"`
 
+	// Device type, one of: HDD, SSD, NVME
```

Comment:
Since we're interpreting discovered data to report the type, we should include the rules here in this comment since it's the main documentation a user will see via the CRD.

Reply:
Yea, seems reasonable, will do.